tests/metrics_test.py

```diff
@@ -137,33 +138,163 @@ def raise_unknown_error():
         self.assertEqual("An unknown error occurred while running the operation [raise_unknown_error] against your Elasticsearch metrics "
                          "store on host [127.0.0.1] at port [9243].", ctx.exception.args[0])
 
-    def test_retries_on_timeouts(self):
-        gateway_timeout = elasticsearch.exceptions.TransportError(504, "Gateway timeout")
+    @mock.patch("esrally.time.sleep")
+    def test_retries_on_bad_gateway(self, mocked_sleep):
+        bad_gateway = elasticsearch.exceptions.TransportError(502, "Bad Gateway")
 
         # should return on first success
-        operation = mock.Mock(side_effect=[gateway_timeout, gateway_timeout, "success", gateway_timeout])
+        operation = mock.Mock(side_effect=[bad_gateway, bad_gateway, "success", bad_gateway])
```

Comment:
Maybe randomize and vary the error type here? I think then we could also reduce the number of individual test cases?